### PR TITLE
disable nullable support for clickhouse destination

### PR DESCRIPTION
**Pull request by gane5hvarma**
This change is [Reviewable](https://reviewable.io/reviews/rudderlabs/rudder-server/746)

### CODE REVIEW

File: warehouse/clickhouse/clickhouse.go [hunk 1]
```diff
@@ -187,30 +190,35 @@ func (ch *HandleT) getConnectionCredentials() credentialsT {
...
 func getClickHouseColumnTypeForSpecificTable(tableName string, columnType string, notNullableKey bool) string {
 	if notNullableKey {
 		return columnType
 	}
+	// Nullable is not disabled for users and identity table
 	if tableName == warehouseutils.UsersTable {
 		return fmt.Sprintf(`SimpleAggregateFunction(anyLast, Nullable(%s))`, columnType)
 	}
+	if tableName != warehouseutils.IdentifiesTable && disableNullable {
+		return fmt.Sprintf(`%s`, columnType)
```
Comment (sivashanmukh) [inline warehouse/clickhouse/clickhouse.go:220]: Remove fmt.Sprintf() and directly return

File: warehouse/clickhouse/clickhouse.go [hunk 2]
```diff
@@ -270,28 +278,41 @@ func generateArgumentString(arg string, length int) string {
 
 // typecastDataFromType typeCasts string data to the mentioned data type
 func typecastDataFromType(data string, dataType string) interface{} {
+
 	switch dataType {
 	case "int":
 		i, err := strconv.Atoi(data)
 		if err != nil {
+			if disableNullable {
```
Comment (sivashanmukh) [inline warehouse/clickhouse/clickhouse.go:286]: Maintain a map[string]interface{} where key is dataType and value is the default value for that type.

Let the error be caught after switch and set the value corresponding to that dataType on error.

**Review by sivashanmukh — APPROVED** [review]
LGTM

File: warehouse/clickhouse/clickhouse.go [hunk 3]
```diff
@@ -187,30 +197,35 @@ func (ch *HandleT) getConnectionCredentials() credentialsT {
...
 
-func getClickHouseCodecForColumnType(columnType string) string {
+func getClickHouseCodecForColumnType(columnType string, tableName string) string {
 	switch columnType {
 	case "datetime":
-		return "Codec(DoubleDelta, LZ4)"
-	default:
-		return ""
+		if disableNullable && !(tableName == warehouseutils.IdentifiesTable || tableName == warehouseutils.UsersTable) {
+			return "Codec(DoubleDelta, LZ4)"
```
Comment (psrikanth88) [inline warehouse/clickhouse/clickhouse.go:211]: @gane5hvarma are we adding compression only for datetime fields?

**Review by psrikanth88 — APPROVED** [review]

**Pull request merged**